pootle/apps/pootle_app/management/commands/update_external_tmserver.py

```diff
@@ -0,0 +1,151 @@
+                         "created on the server if it doesn't exist"),
+        make_option('--target-language',
+                    action='store',
+                    dest='target_language',
+                    help="Target language to fallback to use in case it can't "
+                         "be guessed for any of the input files."),
+    )
+
+    def _parse_translations(self, *args, **options):
+        units = []
+
+        for filename in args:
+            store = factory.getobject(filename)
+            if (not store.gettargetlanguage() and
+                not self.target_language and
+                not options['dry_run']):
+
+                raise CommandError("Unable to get target language for '%s'. "
+                                   "Please specify a fallback target language "
+                                   "with --target-language" % filename)
+            units.extend([unit for unit in store.units if unit.istranslated()])
+
+        total = len(units)
+
+        if total == 0:
+            self.stdout.write("No translations to index")
+            sys.exit()
+
+        self.stdout.write("%s translations to index" % total)
+
+        if options['dry_run']:
+            sys.exit()
+
+        self.stdout.write("")
+
+        for i, unit in enumerate(units, start=1):
+            if (i % 1000 == 0) or (i == total):
+                percent = "%.1f" % (i * 100.0 / total)
+                self.stdout.write("%s (%s%%)" % (i, percent), ending='\r')
+                self.stdout.flush()
+
+            target_language = unit.gettargetlanguage()
+            if target_language is None:
+                target_language = self.target_language
+
+            yield {
+                "_index": self.INDEX_NAME,
+                "_type": target_language,
+                "_id": unit.getid(),
+                'revision': 0,
+                'project': self.tm_settings['TM_NAME'],
+                'path': filename,
+                'username': None,
+                'fullname': None,
+                'email_md5': None,
+                'source': unit.source,
+                'target': unit.target,
+            }
+
+    def handle(self, *args, **options):
+
+        if not getattr(settings, 'POOTLE_TM_SERVER', False):
+            raise CommandError('POOTLE_TM_SERVER setting is missing.')
+
+        if not len(args):
+            raise CommandError('Some file to import must be specified.')
+
+        self.target_language = options.pop('target_language', '')
+        tm_server = options.pop('tm_server', '')
+
+        if not tm_server:
+            # Use default external TM.
+            tm_server = 'external'
+
+        try:
+            self.tm_settings = settings.POOTLE_TM_SERVER[tm_server]
+        except KeyError:
+            raise CommandError('Specified TM is not on POOTLE_TM_SERVER.')
+
+        if 'IS_LOCAL' not in self.tm_settings:
+            raise CommandError('IS_LOCAL missing for specified TM server.')
```

Comment:
It is necessary to tell if the server is either local or external, so when running this command we don't inadvertently push translations to a local TM.